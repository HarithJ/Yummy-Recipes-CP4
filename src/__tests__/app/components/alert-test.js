import React from 'react';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson, { shallowToJson } from 'enzyme-to-json';

import Alert from '../../../components/alert.js';

describe("alert", () => {
  describe("alert msg ", () => {
    it("should display the name of the category", () => {
      const alert = shallow(<Alert alertType={"alert-success"} alertMsg={"A message to test testing"}/>)
      const alertMsg = alert.find('div').text();
      expect(alertMsg).toEqual('A message to test testing')
    })

    it("renders properly", () => {
      const alert = shallow(<Alert alertType={"alert-success"} alertMsg={"A message to test testing"}/>)
      expect(shallowToJson(alert)).toMatchSnapshot();
    })

    it("has a function to hide alert", () => {
      const alert = shallow(<Alert alertType={"alert-success"} alertMsg={"A message to test testing"}/>)

      alert.setProps({hideAlert: jest.fn()})

      expect(alert.instance().hide());
    })
  })
})
