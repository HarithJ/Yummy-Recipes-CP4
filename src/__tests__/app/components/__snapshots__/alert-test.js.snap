// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert alert msg  renders properly 1`] = `
<div
  className="text-center alert alert-success"
  role="alert"
>
  A message to test testing
</div>
`;
